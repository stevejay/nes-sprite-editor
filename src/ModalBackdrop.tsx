import React from "react";
import styles from "./ModalBackdrop.module.scss";

type Props = {
  opacity: number;
};

const ModalBackdrop: React.FunctionComponent<Props> = ({ opacity }) => (
  <div className={styles.backdrop} tabIndex={-1} style={{ opacity }} />
);

export default ModalBackdrop;
