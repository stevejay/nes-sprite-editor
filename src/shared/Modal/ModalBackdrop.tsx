import React from "react";
import styles from "./ModalBackdrop.module.scss";

type Props = {
  opacity: number;
  onClose?: () => void;
};

const ModalBackdrop = ({ opacity, onClose }: Props) => {
  const eventHandler = (event: React.SyntheticEvent) => {
    event.preventDefault();
    onClose && onClose();
  };

  return (
    <div
      className={styles.backdrop}
      tabIndex={-1}
      style={{ opacity }}
      onMouseDown={onClose ? eventHandler : undefined}
      onTouchStart={onClose ? eventHandler : undefined}
    />
  );
};

export default React.memo(ModalBackdrop);
