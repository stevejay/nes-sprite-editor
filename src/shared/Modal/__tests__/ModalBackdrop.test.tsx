import React from "react";
import { render, cleanup } from "react-testing-library";
import userEvent from "user-event";
import ModalBackdrop from "../ModalBackdrop";

afterEach(cleanup);

test("displays correctly", async () => {
  const { container } = render(<ModalBackdrop opacity={0.5} />);
  expect(container).toMatchSnapshot();
});

test("reacts to mouse click", async () => {
  const handleClose = jest.fn();
  const { container } = render(
    <ModalBackdrop opacity={0.5} onClose={handleClose} />
  );
  userEvent.click(container.children[0]);
  expect(handleClose).toHaveBeenCalledTimes(1);
});
