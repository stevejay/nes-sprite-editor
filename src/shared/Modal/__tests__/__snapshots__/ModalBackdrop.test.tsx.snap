// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays correctly 1`] = `
<div>
  <div
    class="backdrop"
    style="opacity: 0.5;"
    tabindex="-1"
  />
</div>
`;
